crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -67,9 +67,14 @@ impl MsgArea {
     }
 
     pub(crate) fn resize(&mut self, width: i32, height: i32) {
-        self.width = width;
+        self.width = if self.scroll != 0 { width - 1 } else { width };
+        let old_height = self.height;
         self.height = height;
+        let old_total_lines = self.lines_height;
```

Comment:
Not in this PR but we should probably rename `lines_height` to `total_lines` or something else that describes it better. The fact that you used `old_total_lines` and not `old_lines_height` suggests `line_height` is not a good name for it.